modules/layers/src/solid-polygon-layer/polygon.js

```diff
@@ -38,6 +38,21 @@ export function isSimple(polygon) {
   return polygon.length >= 1 && polygon[0].length >= 2 && Number.isFinite(polygon[0][0]);
 }
 
+/**
+ * Ensure that all simple polygons have the same start and end vertex
+ */
+function closeLoop(simplePolygon) {
+  // check if first and last vertex are the same
+  const p0 = simplePolygon[0];
+  const p1 = simplePolygon[simplePolygon.length - 1];
+
+  if (p0[0] === p1[0] && p0[1] === p1[1] && p0[2] === p1[2]) {
+    return simplePolygon;
+  }
+  // duplicate the starting vertex at end
+  return simplePolygon.concat([p0]);
```

Comment:
I have this general vision that we wouldn't modify or transform incoming geojson data. Both from an immutability perspective, but also thinking that any additions/changes we need should be done in the buffers we are building, not in the source data.

But I suspect that I would not like the resulting complexity...

Reply:
`concat` does not modify the original array.